src/App.tsx

```diff
@@ -1,16 +1,80 @@
 import React from 'react';
 import './App.scss';
 
-class App extends React.Component {
-  addOne = () => {};
+type State = {
+  sum: number,
+};
 
-  add100 = () => {};
+type Props = {
 
-  increase = () => {};
+};
+
+class App extends React.Component<Props, State> {
+  state: State = {
+    sum: 0,
+  };
+
+  addOne = () => {
+    this.setState((state: State) => ({
+      sum: state.sum + 1,
+    }));
+  };
+
+  add100 = () => {
+    this.setState((state: State) => ({
+      sum: state.sum + 100,
+    }));
+  };
+
+  increase = () => (
+    this.state.sum % 5 ? (
+      this.addOne()
+    ) : (
+      this.add100()
+    )
+  );
 
   render() {
+    const { sum } = this.state;
+
     return (
-      <h1>Count: 0</h1>
+      <div>
+        <h1>
+          Count:
+          {sum}
+        </h1>
+
+        <div>
+          <button
+            type="button"
+            onClick={() => {
+              this.addOne();
+            }}
```

Comment:
```suggestion
            onClick={this.addOne}
```
you can remove the arrow function because method addOne doesn't accept any arguments